source/extensions/filters/http/ext_authz/config.cc

```diff
@@ -52,13 +53,16 @@ Http::FilterFactoryCb ExtAuthzFilterConfig::createFilterFactoryFromProtoTyped(
 
     const uint32_t timeout_ms =
         PROTOBUF_GET_MS_OR_DEFAULT(proto_config.grpc_service(), timeout, DefaultTimeout);
-    auto async_client_cache = getAsyncClientCacheSingleton(context);
+    auto async_client_cache_singleton = Grpc::getAsyncClientCacheSingleton(context);
+    auto async_client_cache = async_client_cache_singleton->getOrCreateAsyncClientCache(
```

Comment:
IMO it would be easier to read if you use explicit types here rather than `auto`, as the types are not lexically obvious from context, and to understand this code, it matters that they are shared pointers.